addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -2578,6 +2578,6 @@ export function isMacOS() {
  */
 export function cleanZWS(node) {
     [node, ...descendants(node)]
-        .filter(node => node.nodeType === Node.TEXT_NODE)
+        .filter(node => node.nodeType === Node.TEXT_NODE && node.nodeValue.includes('\u200B'))
```

Comment:
It broke the selection in the tests if we arbitrarily changed the node value of the node we were in.

Reply:
Oh ok I see, thanks ! 👍

Reply:
```
[FIX] web_editor: only clean ZWS in text nodes actually containing ZWS

When cleaning zero-width spaces (ZWS) in the editor, we were replacing
the node values of all text nodes descending from the node we were
cleaning, regardless of whether they contained a ZWS or not. This could
lead to wrong selections in the tests when arbitrarily changing the
value of a text node in the selection.
```